servicetalk-opentracing-zipkin-publisher/src/test/java/io/servicetalk/opentracing/zipkin/publisher/reporter/HttpReporterTest.java

```diff
@@ -0,0 +1,165 @@
+import static io.servicetalk.transport.netty.internal.AddressUtils.serverHostAndPort;
+import static java.time.Duration.ofMillis;
+import static java.util.Arrays.asList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertThrows;
+
+@RunWith(Parameterized.class)
+public class HttpReporterTest {
+    @Rule
+    public final Timeout timeout = new ServiceTalkTestTimeout();
+
+    private final BlockingQueue<HttpRequest> receivedRequests;
+    private final ServerContext context;
+    private final Codec codec;
+    @Nullable
+    private HttpReporter reporter;
+
+    public HttpReporterTest(final Codec codec) throws Exception {
+        this.codec = codec;
+        receivedRequests = new LinkedBlockingQueue<>();
+        this.context = forAddress(localAddress(0))
+                .listenBlockingAndAwait((ctx, request, responseFactory) -> {
+                    receivedRequests.add(request);
+                    return responseFactory.ok();
+                });
+    }
+
+    @Parameterized.Parameters(name = "codec: {0}")
+    public static Collection<Codec> data() {
+        return asList(Codec.values());
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        CompositeCloseable closeable = AsyncCloseables.newCompositeCloseable();
+        if (reporter != null) {
+            closeable.append(reporter);
+        }
+        closeable.append(context);
+        closeable.closeGracefully();
+    }
+
+    @Test
+    public void disableBatching() throws Exception {
+        HttpReporter reporter = initReporter(Builder::disableSpanBatching);
+        reporter.report(newSpan("1"));
+        List<Span> spans = verifyRequest(receivedRequests.take(), false);
+        assertThat("Unexpected spans received.", spans, hasSize(1));
+        verifySpan(spans.get(0), "1");
+
+        reporter.report(newSpan("2"));
+        List<Span> spans2 = verifyRequest(receivedRequests.take(), false);
+        assertThat("Unexpected spans received.", spans, hasSize(1));
+        verifySpan(spans2.get(0), "2");
+    }
+
+    @Test
+    public void batching() throws Exception {
+        HttpReporter reporter = initReporter(builder -> builder.batchSpans(2, ofMillis(200)));
+        reporter.report(newSpan("1"));
+        reporter.report(newSpan("2"));
+        List<Span> spans = verifyRequest(receivedRequests.take(), true);
+        assertThat("Unexpected spans received.", spans, hasSize(2));
+        verifySpan(spans.get(0), "1");
+        verifySpan(spans.get(1), "2");
+    }
+
+    @Test
+    public void reportAfterClose() {
+        HttpReporter reporter = initReporter(Builder::disableSpanBatching);
+        reporter.close();
+        assertThrows("Report post close accepted.", IllegalStateException.class,
+                () -> reporter.report(newSpan("1")));
```

Comment:
Consider also an assert for `reporter.check()` before this one.

Reply:
Good call, there was a bug 🤕